scheduler/leader.go

```diff
@@ -70,12 +70,18 @@ func Lead(shutdown <-chan interface{}, conn coordclient.Connection, cpClient dao
 // service has an address assignment the host will already be selected. If not
 // the host with the least amount of memory committed to running containers will
 // be chosen.  Returns the hostid, hostip (if it has an address assignment).
-func (l *leader) SelectHost(s *service.Service) (string, error) {
+func (l *leader) SelectHost(sn *zkservice.ServiceNode) (string, error) {
```

Comment:
I could add the fields to ServiceNode, but we'd still need to get the service to populate the Endpoints for use here: https://github.com/control-center/serviced/blob/e290a689473e028c9e78d7b06e6dc5d81f5e717f/scheduler/leader.go#L97-L108